File: src/bindings/python/DocStrings/CDLTransform.py

class CDLTransform:
    """
    CDLTransform
    """
    def __init__(self):
        pass
    
    def CreateFromFile(self, src, cccid):
        pass
    
    def equals(self, cdl):
        pass
        
    def getXML(self):
        pass
        
    def setXML(self, xmltext):
        pass
        
    def getSlope(self):
        pass
        
    def getOffset(self):
        pass
        
    def getPower(self):
        pass
        
    def getSOP(self):
        pass
        
    def getSat(self):
        pass
        
    def setSlope(self, slope):
        """
        setSlope(pyData)
        
        Sets the slope ('S' part of SOP) in :py:class:`PyOpenColorIO.CDLTransform`.
        
        :param pyData: 
        :type pyData: object
        """
        pass
        
    def setOffset(self, offset):
        """
        setOffset(pyData)
        
        Sets the offset ('O' part of SOP) in :py:class:`PyOpenColorIO.CDLTransform`.
        
        :param pyData: list of three floats
        :type pyData: object
        """
        pass
        
    def setPower(self, power):
        """
        setPower(pyData)
        
        Sets the power ('P' part of SOP) in :py:class:`PyOpenColorIO.CDLTransform`.
        
        :param pyData: list of three floats
        :type pyData: object
        """
        pass

    def setSOP(self, sop):
        """
        setSOP(pyData)
        
        Sets SOP in :py:class:`PyOpenColorIO.CDLTransform`.
        
        :param pyData: list of nine floats
        :type pyData: object
        """
        pass
        
    def setSat(self, sat):
        """
        setSAT(pyData)
        
        Sets SAT (saturation) in :py:class:`PyOpenColorIO.CDLTransform`.
        
        :param pyData: saturation
        :type pyData: float
        """
        pass
        
    def getSatLumaCoefs(self):
        """
        getSatLumaCoefs(pyData)
        
        Returns the SAT (saturation) and luma coefficients in :py:class:`PyOpenColorIO.CDLTransform`.
        
        :return: saturation and luma coefficients
        :rtype: list of floats
        """
        pass
        
    def getID(self):
        """
        getID()
        
        Returns the ID from :py:class:`PyOpenColorIO.CDLTransform`.
        
        :return: ID
        :rtype: string
        """
        pass
        
    def setID(self, id):
        """
        setID(str)
        
        Sets the ID in :py:class:`PyOpenColorIO.CDLTransform`.
        
        :param str: ID
        :type str: string
        """
        pass
        
    def getDescription(self):
        """
        getDescription()
        
        Returns the description of :py:class:`PyOpenColorIO.CDLTransform`.
        
        :return: description
        :rtype: string
        """
        pass
        
    def setDescription(self, desc):
        """
        setDescription(str)
        
        Sets the description of :py:class:`PyOpenColorIO.CDLTransform`.
        
        :param str: description
        :type str: string
        """
        pass

File: src/bindings/java/org/OpenColorIO/CDLTransform.java
// SPDX-License-Identifier: BSD-3-Clause
// Copyright Contributors to the OpenColorIO Project.

package org.OpenColorIO;
import org.OpenColorIO.*;

public class CDLTransform extends Transform
{
    public CDLTransform() { super(); }
    protected CDLTransform(long impl) { super(impl); }
    public native CDLTransform Create();
    public native CDLTransform CreateFromFile(String src, String cccid);
    public native boolean equals(CDLTransform obj);
    public native String getXML();
    public native void setXML(String xml);
    public native void setSlope(float[] rgb);
    public native void getSlope(float[] rgb);
    public native void setOffset(float[] rgb);
    public native void getOffset(float[] rgb);
    public native void setPower(float[] rgb);
    public native void getPower(float[] rgb);
    public native void setSOP(float[] vec9);
    public native void getSOP(float[] vec9);
    public native void setSat(float sat);
    public native float getSat();
    public native void getSatLumaCoefs(float[] rgb);
    public native void setID(String id);
    public native String getID();
    public native void setDescription(String desc);
    public native String getDescription();
}
